lib/interpreters/split_clustering.rb

```diff
@@ -129,61 +120,306 @@ def self.generate_split_vrps(service_vrp, _job = nil, block = nil)
 
-        sub_vrp.resolution_duration = vrp.resolution_duration&.*(sub_problem_size_ratio)&.+(transfer_unused_time_limit)&.ceil
-        sub_vrp.resolution_minimum_duration = (vrp.resolution_minimum_duration || vrp.resolution_initial_time_out)&.*(sub_problem_size_ratio)&.ceil
-        sub_vrp.resolution_iterations_without_improvment = vrp.resolution_iterations_without_improvment&.*(sub_problem_size_ratio)&.ceil
-        sub_vrp.resolution_vehicle_limit = [(vrp.resolution_vehicle_limit || vrp.vehicles.size)&.*(sub_problem_size_ratio)&.round(0)&.+(transfer_unused_vehicle_limit), 1].max
+          vehicle_limit_ratio = current_vehicle_limit.to_f * side.size / enum_current_vehicles.size
+          # Warning: round does not work if there is an even "half" split
+          ss_data[:current_vehicle_limit] = current_vehicle_limit &&
+            (index.zero? ? vehicle_limit_ratio.ceil : vehicle_limit_ratio.floor)
 
-        sub_problem = {
-          vrp: sub_vrp,
-          service: service_vrp[:service]
+          split_solve_core(service_vrp, job = nil, &block)
         }
-        sub_vrp.services += empties_or_fills
-        sub_vrp.points += empties_or_fills.map{ |empti_of_fill| vrp.points.find{ |point| empti_of_fill.activity.point.id == point.id } }
-        sub_vrp.vehicles.select!{ |vehicle| available_vehicle_ids.include?(vehicle.id) }
+        log "<-- split_solve_core level: #{split_level}"
+
+        Helper.merge_results(results)
+      else # Stopping condition -- the problem is small enough
+        # Finally generate the sub-vrp without hard-copy and solve it
+        result = split_solve_sub_vrp(service_vrp, job, &block)
+
+        transfer_unused_resources(ss_data, service_vrp[:vrp], result)
+
+        result
+      end
+    end
+
+    def self.split_solve_sub_vrp(service_vrp, job = nil, &block)
+      ss_data = service_vrp[:split_solve_data]
+      split_level = service_vrp[:split_level] # local split_level var is needed to show the level value correctly
+      service_cnt = "#{ss_data[:current_vehicles].sum{ |v| ss_data[:service_vehicle_assignments][v.id].size }} services"
+      vehicle_cnt = "#{ss_data[:current_vehicles].size}+#{ss_data[:transferred_vehicles].size} vehicles"
+      log "--> split_solve_sub_vrp lv: #{split_level} | solving #{service_cnt} with #{vehicle_cnt}"
+
+      service_vrp[:vrp] = create_sub_vrp(ss_data)
+
+      OptimizerWrapper.define_process(service_vrp, job, &block)
+    ensure
+      log "<-- split_solve_sub_vrp lv: #{split_level}"
+    end
+
+    def self.create_sub_vrp(split_solve_data)
+      ss_data = split_solve_data
+      o_vrp = ss_data[:original_vrp]
+
+      sub_vrp = ::Models::Vrp.create({}, false)
+
+      # Select the vehicles and services belonging to this sub-problem from the service_vehicle_assignments
+      sub_vrp.vehicles = ss_data[:current_vehicles] + ss_data[:transferred_vehicles]
+      sub_vrp.vehicles.each{ |v| v.cost_fixed = 1 if v.cost_fixed == 0 }
```

Comment:
Do you set cost_fixed to 1 to avoid dicho ? Can't we call it after split_solve, to split really big problems and avoid requesting large matrices ?